app/assets/javascripts/orders.coffee

```diff
@@ -81,6 +81,7 @@ addListeners = (element) ->
     populateItems @value, items
 
 $(document).on 'page:change', ->
-  element = findLastCategory()
-  populateCategories element
-  addListeners element
+  if typeof data != "undefined"
```

Comment:
This will trigger on every page change, meaning after the orders page is hit, the populating categories code will hit on every single page (since the data will now be defined). I think this would be better served as an inline javascript tag just after the categories element, like: `<%= javascript_tag "populateCategories(findLastCategory())" %>` This might require defining these functions like `window.populateCategories = -> ...` (since coffeescript is usually compiled within a scope, restricting what gets defined in the global context to things define explicitly in the global context).

Also, it seems like you should be able to simply do `<%= javascript_tag "populateCategories($("#category"))" %>` since it is an element id you are looking up.

I think the listeners don't really need to be hooked up like this... instead it could be a document level listener, like most others in StockAid, like: `$(document).on "change", "#category", -> ...`

Also, in the addListeners code, I noticed `items = $(event.currentTarget.parentElement.parentElement).find '#item'`. Depending on element structure like this is extremely fragile, as now a slight change in the layout could easily break the code. Instead, it would be better to do something like: `$(event.currentTarget).parents(".some-specific-parent-class:first").find(".some-child")`. However, since this case is finding by an id, you can instead simply do `$("#item")` because element ids are unique... there is no need to find an element by id scoped to something. Elements cannot have the same id (or if they can, it is extremely sensitive to different browsers and different browser versions, so generally speaking, ids should be never be defined with duplicate values).